ghproxy/ghmetrics/ghpath.go

```diff
@@ -0,0 +1,424 @@
+							children: []node{{
+								Node: variable("sha"),
+							}},
+						},
+						{
+							Node: literal("git"),
+							children: []node{
+								{
+									Node: literal("commits"),
+									children: []node{{
+										Node: variable("sha"),
+									}},
+								},
+								{
+									Node: literal("ref"),
+									children: []node{{
+										Node: variable("refId"),
+									}},
+								},
+								{
+									Node: literal("tags"),
+									children: []node{{
+										Node: variable("tagId"),
+									}},
+								},
+								{
+									Node: literal("trees"),
+									children: []node{{
+										Node: variable("sha"),
+									}},
+								},
+							},
+						},
+						{Node: literal("stars")},
+						{Node: literal("merges")},
+						{Node: literal("stargazers")},
+						{Node: literal("notifications")},
+						{Node: literal("hooks")},
+						{Node: literal("deployments")},
+						{Node: literal("downloads")},
+						{Node: literal("events")},
+						{Node: literal("forks")},
+						{Node: literal("topics")},
+						{Node: literal("vulnerability-alerts")},
+						{Node: literal("automated-security-fixes")},
+						{Node: literal("contributors")},
+						{Node: literal("languages")},
+						{Node: literal("teams")},
+						{Node: literal("tags")},
+						{Node: literal("transfer")},
+					},
+				}},
+			}},
+		}
+
+	case "user":
+		// /user
+		tree = node{
+			Node: literal("user"),
+			children: []node{
+				{
+					Node: literal("following"),
+					children: []node{{
+						Node: variable("userId"),
+					}},
+				},
+				{
+					Node: literal("keys"),
+					children: []node{{
+						Node: variable("keyId"),
+					}},
+				},
+				{
+					Node: literal("email"),
+					children: []node{{
+						Node: literal("visibility"),
+					}},
+				},
+				{Node: literal("emails")},
+				{Node: literal("public_emails")},
+				{Node: literal("followers")},
+				{Node: literal("starred")},
+				{Node: literal("issues")},
+			},
+		}
+	case "users":
+		// /users
+		tree = node{
+			Node: literal("users"),
+			children: []node{{
+				Node: variable("username"),
+				children: []node{
+					{
+						Node: literal("followers"),
+						children: []node{{
+							Node: variable("username"),
+						}},
+					},
+					{Node: literal("repos")},
+					{Node: literal("hovercard")},
+					{Node: literal("following")},
+				},
+			}},
+		}
+	case "orgs":
+		// /orgs
+		tree = node{
+			Node: literal("orgs"),
+			children: []node{{
+				Node: variable("orgname"),
+				children: []node{
+					{
+						Node: literal("credential-authorizations"),
+						children: []node{{
+							Node: variable("credentialId"),
+						}},
+					},
+					{Node: literal("repos")},
+					{Node: literal("issues")},
+				},
+			}},
+		}
+	case "issues":
+		// /issues
+		tree = node{
+			Node: literal("issues"),
+			children: []node{{
+				Node: variable("issueId"),
+			}},
+		}
+	case "search":
+		// do we care to handle search sub-paths differently?
```

Comment:
Unless someone else explicitly says "yes", the answer is probably "no" to all of these.

Logging that there were "X searches for Y" seems sufficient to me.